### PR TITLE
Support multiple -*notify commands

### PR DESCRIPTION
**Pull request by luke-jr**
Useful, intuitive, and avoids creating issues with #22354

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -939,29 +939,31 @@ CWalletTx* CWallet::AddToWallet(CTransactionRef tx, const CWalletTx::Confirmatio
...
-        }
-#ifndef WIN32
-        // Substituting the wallet name isn't currently supported on windows
-        // because windows shell escaping has not been implemented yet:
-        // https://github.com/bitcoin/bitcoin/pull/13339#issuecomment-537384875
-        // A few ways it could be implemented in the future are described in:
-        // https://github.com/bitcoin/bitcoin/pull/13339#issuecomment-461288094
-        boost::replace_all(strCmd, "%w", ShellEscape(GetName()));
+    if (gArgs.IsArgSet("-walletnotify")) {
+#ifdef WIN32
```
Comment (kristapsk) [inline src/wallet/wallet.cpp:943]: Comment below `#ifndef WIN32` from previous code about why it's not supported currently should be copied here IMO.

**Comment by kristapsk** [comment]
Concept ACK

**Comment by benthecarman** [comment]
Concept ACK

**Comment by benthecarman** [comment]
Looks like this doesn't touch -startupnotify, does that support multiple already?

**Comment by luke-jr** [comment]
Whoops, fixed.

File: src/init.cpp [hunk 2]
```diff
@@ -683,9 +683,8 @@ static void CleanupBlockRevFiles()
 #if HAVE_SYSTEM
 static void StartupNotify(const ArgsManager& args)
 {
-    std::string cmd = args.GetArg("-startupnotify", "");
-    if (!cmd.empty()) {
-        std::thread t(runCommand, cmd);
+    for (std::string command : gArgs.GetArgs("-startupnotify")) {
```
Comment (MarcoFalke) [inline src/init.cpp:686]:
```suggestion
    for (std::string cmd : args.GetArgs("-startupnotify")) {
```

Seems odd to use gArgs when args is passed in. Also, why change the symbol name to increase the diff?

Comment (luke-jr) [reply src/init.cpp]: gArgs: Fixed.

Symbol name: It doesn't increase the diff to adjust it to our current style.

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (luke-jr) [reply src/wallet/wallet.cpp:996]: Copied

**Comment by luke-jr** [comment]
Rebased and addressed review comments

**Comment by achow101** [comment]
Closing this as it has not had any activity in a while. If you are interested in continuing work on this, please leave a comment so that it can be reopened.

**Comment by kristapsk** [comment]
Guess this was closed just because of no rebase after May 6 and lack of ACKs? Seems useful in context of #25975.

**Comment by luke-jr** [comment]
Feel free to reopen. As with most of my PRs, I continue to maintain it for Knots.

**Pull request closed without merging**